plugins/providers/virtualbox/action/import_master.rb

```diff
@@ -0,0 +1,62 @@
+require "log4r"
+#require "lockfile"
+
+module VagrantPlugins
+  module ProviderVirtualBox
+    module Action
+      class ImportMaster
+        def initialize(app, env)
+          @app = app
+          @logger = Log4r::Logger.new("vagrant::action::vm::create_master")
+        end
+
+        def call(env)
+          master_id_file = env[:machine].box.directory.join("master_id")
+                    
+          env[:master_id] = master_id_file.read.chomp if master_id_file.file?          
+          if env[:master_id] && env[:machine].provider.driver.vm_exists?(env[:master_id])
+            # Master VM already exists -> nothing to do - continue.
+            @app.call(env) 
+          end          
+          
+          env[:ui].info I18n.t("vagrant.actions.vm.clone.importing", name: env[:machine].box.name)
+            
+          #TODO - prevent concurrent creation of master vms for the same box.

```

Comment:
I think we will want to handle this before we can merge. The easiest solution (although not the most performant) is to lock everything up to line 16 in the global lock. That way a master is created one at a time.